streams/test-utils/src/test/java/org/apache/kafka/streams/TopologyTestDriverTest.java

```diff
@@ -711,6 +715,56 @@ public void shouldUseSourceSpecificDeserializers() {
         assertThat(result2.getValue(), equalTo(source2Value));
     }
 
+    @Test
+    public void shouldPassRecordHeadersIntoDeSerializers() {
```

Comment:
It seems like we are just verifying the headers are deserialized. Should we just use mock deserializers? Moreover, we could build a customized `boolean` deserializer which outputs true when the header is provided, or false otherwise.

Reply:
We don't verify that header are deserialized -- we just verify that the right function is called, that hands the header into the deserializer (ie, to allow used to read them as metadata).

I guess we should use mocks for the deserializers, but not for the serializers. Thus, not using mocks make the test use the same patter for both what seems "nicer".

> Moreover, we could build a customized boolean deserializer which outputs true when the header is provided, or false otherwise.

Would this make the test simpler? I personally doubt it.